WalletWasabi/Crypto/ZeroKnowledge/NonInteractive/IKnowledge.cs

```diff
@@ -0,0 +1,21 @@
+using WalletWasabi.Crypto.Randomness;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge.NonInteractive
+{
+	// IKnowledge splits proving into 3 phases
+	// The second phase is to generate and commit to all the nonces
```

Comment:
```suggestion
	/// <summary>
	/// <see cref="IKnowledge"/> splits proving into 3 phases.
	/// The second phase is to generate and commit to all the nonces.
	/// </summary>
```

xmldoc is a standard way in C# projects to annotate methods, fields, classes and namespaces. Your comment may be obsolete if somebody renames `IKnowledge` interface.

Reply:
Also, it would be nice to end sentences with dots.

Reply:
Also this comment should probably be on the interface itself.